kernel/time/src/lib.rs

```diff
@@ -44,7 +31,11 @@ pub struct Instant {
 impl Instant {
     const ZERO: Self = Self { counter: 0 };
 
-    /// Returns the amout of time elapsed from another instant to this one, or
+    pub fn new(counter: u64) -> Self {
+        Self { counter }
+    }
+
+    /// Returns the amount of time elapsed from another instant to this one, or
     /// zero duration if that instant is later than this one.
     pub fn duration_since(&self, earlier: Self) -> Duration {
```

Comment:
kindly deduplicate this function. Something like `checked_duration_since(...).unwrap_or(Duration::ZERO)`